compiler/ml/typecore.ml

```diff
@@ -3034,7 +3034,7 @@ and type_expect_ ?type_clash_context ?in_function ?(recarg = Rejected) env sexp
         (Error
            (e.pexp_loc, env, Undefined_method (obj.exp_type, met, valid_methods)))
     )
-  | Pexp_new _ | Pexp_setinstvar _ | Pexp_override _ -> assert false
+  | Pexp_setinstvar _ | Pexp_override _ -> assert false
```

Comment:
@nojaf would you remove these 2 as well, they go well together

Reply:
Sure thing. Is `Pexp_poly` used? Found similar comments there as well.

Reply:
@cristianoc CI seems to agree with it, let me know if we still need to keep it.